sys/net/gnrc/network_layer/ipv6/nib/nib_ft.c

```diff
@@ -132,10 +132,22 @@ bool gnrc_ipv6_nib_ft_iter(const ipv6_addr_t *next_hop, unsigned iface,
 
         while ((offl = _nib_offl_iter(offl))) {
             assert(offl->mode != 0);
-            if ((offl->next_hop != NULL) &&
-                ((iface == 0) || (iface == _nib_onl_get_if(offl->next_hop))) &&
-                ((next_hop == NULL) || ipv6_addr_equal(&offl->next_hop->ipv6,
-                                                       next_hop))) {
+            if ((offl->next_hop == NULL) {
+                /* ??? */
```

Comment:
heh I hoped you could fill in that one

Reply:
😅 I did, but I don't know such case [1], so I just left it as-is, as I think the comment suits an unexpected situation. I did not add that condition, but I don't think it conflicts with this PR. It at least serves as a safeguard for 
a few lines later ([`6dbaed6` (#20835)](https://github.com/RIOT-OS/RIOT/pull/20835/commits/6dbaed6d28fb5c35fc89d544710395f02580201f#diff-5ec94a7841970a6246613c19558b24d290794e5105d6a014d3e6e265c07275bfR147)).

[1] https://github.com/RIOT-OS/RIOT/pull/20371#discussion_r1731467175

Reply:
If there is no next hop, it should not show up in the forwarding table, right?

Reply:
I thought that'd be covered by the check for `next_hop`

Reply:
Yeah, I think the `next_hop &&` below can now removed.

Reply:
> I thought that'd be covered by the check for `next_hop`

Or which check for `next_hop` do you mean?

Reply:
> there is no next hop (not even an interface)

Just to clarify: This is never the case with the current code, right? (Just wondering when such condition would be true.)

We can of course still keep it nonetheless, just in case, even for the future, e.g. I guess https://github.com/RIOT-OS/RIOT/pull/20385 could enable that branch.

Reply:
> Just to clarify: This is never the case with the current code, right? (Just wondering when such condition would be true.)

You probably have a much better overview over the code I wrote >5 years ago. I could imagine, however, that this is mainly to satisfy some edge case where the `offl` entry is not deleted yet, while the corresponding `onl` entry is. I propose to dig a little bit into the history of this line using `git blame`.